src/formatDate.js

```diff
@@ -50,7 +51,53 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fF = fromFormat;
+  const tF = toFormat;
+
+  let separator = fF[3];
+  const arrStartYear = date.split(separator);
+  const a = arrStartYear;
+  const arrResult = [];
+  const objDateInfo = {
+    YY: 'YY',
+    MM: 'MM',
+    DD: 'DD',
+    YYYY: 'YYYY',
+  };
+  const o = objDateInfo;
+
+  for (let i = 0; i < fF.length; i++) {
+    switch (fF[i]) {
+      case 'YYYY' :
+        objDateInfo['YYYY'] = arrStartYear[i];
+        objDateInfo['YY'] = arrStartYear[i].slice(-2);
+        break;
+      case 'DD' :
+        objDateInfo['DD'] = arrStartYear[i];
+        break;
+      case 'MM':
+        objDateInfo['MM'] = arrStartYear[i];
+        break;
+      case 'YY':
+        objDateInfo['YY'] = arrStartYear[i];
+        a[i] >= 0 && a[i] <= 29 ? o['YYYY'] = 20 + a[i] : o['YYYY'] = 19 + a[i];
+        break;
+    }
+  }
+
+  for (const char of tF) {
+    for (const key in objDateInfo) {
+      switch (key) {
+        case char:
+          arrResult.push(objDateInfo[key]);
+      }
+    }
+  }
+  separator = tF.slice(-1);
+
+  const dateResult = arrResult.join(separator);
+
+  return (dateResult);
```

Comment:
You don't need any parentheses here
```suggestion
  return dateResult;
```